api/v1alpha1/conditions.go

```diff
@@ -15,3 +15,25 @@ type StardogCondition struct {
 }
 
 type StardogConditionType string
+
+const (
+	// StardogReady tracks if the Stardog has been successfully reconciled.
+	StardogReady StardogConditionType = "Ready"
+	// StardogErrored is given when no connection could be established with the Stardog Database
+	// any other reconciliation error.
+	StardogErrored StardogConditionType = "Errored"
+	// StardogInvalid is given when the the Stardog Spec contains invalid properties. The specification will not be
+	// reconciled.
+	StardogInvalid StardogConditionType = "Invalid"
+	// StardogToBeDeleted is given when the the Stardog Instance has to be deleted. The reconciliation will run until
+	// the Specification is deleted.
+	StardogToBeDeleted StardogConditionType = "ToBeDeleted"
+
+	SyncReasonFailed          = "SynchronizationFailed"
+	SyncReasonSucceeded       = "SynchronizationSucceeded"
+	SyncReasonFailedWithError = "SynchronizationFailedWithError"
+	SyncReasonConfigInvalid   = "InvalidStardogSpec"
+	SyncReasonToBeDeleted     = "ToBeDeletedStardogSpec"
```

Comment:
There should always be an error if reconcile failed, so it's not needed to separate.
```suggestion
	Failed          = "SynchronizationFailed"
	Succeeded       = "SynchronizationSucceeded"
	SpecInvalid   = "InvalidSpec"
	Terminating     = "Terminating"
	PermissionDenied = "PermissionDenied"
```